Content.Server/SurveillanceCamera/Systems/SurveillanceCameraMonitorSystem.cs

```diff
@@ -0,0 +1,505 @@
+using System.Linq;
+using Content.Server.DeviceNetwork;
+using Content.Server.DeviceNetwork.Systems;
+using Content.Server.Power.Components;
+using Content.Server.UserInterface;
+using Content.Server.Wires;
+using Content.Shared.Interaction;
+using Content.Shared.SurveillanceCamera;
+using Robust.Server.GameObjects;
+using Robust.Server.Player;
+
+namespace Content.Server.SurveillanceCamera;
+
+public sealed class SurveillanceCameraMonitorSystem : EntitySystem
+{
+    [Dependency] private readonly SurveillanceCameraSystem _surveillanceCameras = default!;
+    [Dependency] private readonly UserInterfaceSystem _userInterface = default!;
+    [Dependency] private readonly DeviceNetworkSystem _deviceNetworkSystem = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<SurveillanceCameraMonitorComponent, SurveillanceCameraDeactivateEvent>(OnSurveillanceCameraDeactivate);
+        SubscribeLocalEvent<SurveillanceCameraMonitorComponent, BoundUIClosedEvent>(OnBoundUiClose);
+        SubscribeLocalEvent<SurveillanceCameraMonitorComponent, PowerChangedEvent>(OnPowerChanged);
+        SubscribeLocalEvent<SurveillanceCameraMonitorComponent, SurveillanceCameraMonitorSwitchMessage>(OnSwitchMessage);
+        SubscribeLocalEvent<SurveillanceCameraMonitorComponent, DeviceNetworkPacketEvent>(OnPacketReceived);
+        SubscribeLocalEvent<SurveillanceCameraMonitorComponent, SurveillanceCameraMonitorSubnetRequestMessage>(OnSubnetRequest);
+        SubscribeLocalEvent<SurveillanceCameraMonitorComponent, ComponentStartup>(OnComponentStartup);
+        SubscribeLocalEvent<SurveillanceCameraMonitorComponent, AfterActivatableUIOpenEvent>(OnToggleInterface);
+        SubscribeLocalEvent<SurveillanceCameraMonitorComponent, SurveillanceCameraRefreshCamerasMessage>(OnRefreshCamerasMessage);
+        SubscribeLocalEvent<SurveillanceCameraMonitorComponent, SurveillanceCameraRefreshSubnetsMessage>(OnRefreshSubnetsMessage);
+        SubscribeLocalEvent<SurveillanceCameraMonitorComponent, SurveillanceCameraDisconnectMessage>(OnDisconnectMessage);
+    }
+
+    private const float _maxHeartbeatTime = 300f;
+    private const float _heartbeatDelay = 30f;
+    private HashSet<EntityUid> _activeMonitors = new();
+    private List<EntityUid> _toRemove = new();
+
+    public override void Update(float frameTime)
+    {
+        foreach (var uid in _activeMonitors)
+        {
+            if (!TryComp(uid, out SurveillanceCameraMonitorComponent? monitor))
+            {
+                _toRemove.Add(uid);
+                continue;
+            }
```

Comment:
Probably better to just use a dummy "ActiveSurveillanceCameraMonitorComponent" then you can just iterate those components instead. Also as above needs to handle pausing if you don't (TryComp already checks deletion).